app/src/main/java/com/eveningoutpost/dexdrip/ShareModels/ShareRest.java

```diff
@@ -134,15 +156,112 @@ public boolean verify(String hostname, SSLSession session) {
+        //
+
+        // SRMS:
+        //    authenticatePublisherAccount
+        //        StartRemoteMonitoringSession
+        //            checkMonitorAssignment
+        //              !AssignedToYou: updateMonitorAssignment
+        //                  getValidSessionId
+
+        if (sessionId == null || "".equals(sessionId)) {
+
+        }
+        return sessionId;
+    }
+
+    public void getContacts(Callback<List<ExistingFollower>> existingFollowerListener) {
+        dexcomShareApi.getContacts(getSessionId()).enqueue(new AuthenticatingCallback<List<ExistingFollower>>(existingFollowerListener) {
+            @Override
+            void onRetry() {
+                dexcomShareApi.getContacts(getSessionId()).enqueue(this);
+            }
+        });
     }
 
-    public static Map<String, String> querySessionMap(String sessionId) {
-        Map map = new HashMap<String, String>();
-        map.put("sessionID", sessionId);
-        return map;
+    public void uploadBGRecords(final ShareUploadPayload bg, Callback<ResponseBody> callback) {
+        dexcomShareApi.uploadBGRecords(getSessionId(), bg).enqueue(new AuthenticatingCallback<ResponseBody>(callback) {
+            @Override
+            void onRetry() {
+                dexcomShareApi.uploadBGRecords(getSessionId(), bg).enqueue(this);
+            }
+        });
+    }
+
+    public void createContact(final String followerName, final String followerEmail, Callback<String> callback) {
+        dexcomShareApi.createContact(getSessionId(), followerName, followerEmail).enqueue(new AuthenticatingCallback<String>(callback) {
+            @Override
+            void onRetry() {
+                dexcomShareApi.createContact(getSessionId(), followerName, followerEmail).enqueue(this);
+            }
+        });
+    }
+
+    public void createInvitationForContact(final String contactId, final InvitationPayload invitationPayload, Callback<String> callback) {
+        dexcomShareApi.createInvitationForContact(getSessionId(), contactId, invitationPayload).enqueue(new AuthenticatingCallback<String>(callback) {
+            @Override
+            void onRetry() {
+                dexcomShareApi.createInvitationForContact(getSessionId(), contactId, invitationPayload).enqueue(this);
+            }
+        });
+    }
+
+    public void deleteContact(final String contactId, Callback<ResponseBody> deleteFollowerListener) {
+        dexcomShareApi.deleteContact(getSessionId(), contactId).enqueue(new AuthenticatingCallback<ResponseBody>(deleteFollowerListener) {
+            @Override
+            void onRetry() {
+                dexcomShareApi.deleteContact(getSessionId(), contactId).enqueue(this);
+            }
+        });
+    }
+
+    public abstract class AuthenticatingCallback<T> implements Callback<T> {
+
+        int attempts = 0;
+        Callback<T> delegate;
+        public AuthenticatingCallback (Callback<T> callback) {
+            this.delegate = callback;
+        }
+
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1296698395/issues/source?bid=2735053&fileBranchId=2947762#l235)